openmdao/core/system.py

```diff
@@ -2967,10 +2970,12 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
                     out.update(dvs)
 
             if self.comm.size > 1 and self._subsystems_allprocs:
-                allouts = self.comm.allgather(out)
-                out = OrderedDict()
-                for all_out in allouts:
-                    out.update(all_out)
+                for name in out:
```

Comment:
This loop is still a problem.  'out' is the dict that contains dv metadata for ALL dvs, but you're allgathering it for every non-distributed variable.  Instead, you should only do it once like the original code did.  You're basically doing the exact same thing that the old code did except you're doing it multiple times.

Reply:
Yeah I was just about to change that back. I forgot to fix that in the new commit

Reply:
Fixed